tests/System.Text.Primitives.Tests/Encoding/Utf8Utf16ConversionTests.cs

```diff
@@ -78,6 +78,127 @@ public void EnumerateAndEnsureCodePointsOfTheSameUtf8AndUtf16StringsAreTheSame(s
             }
         }
 
+        public static object[][] InvalidUtf8ToUtf16SequenceData = {
```

Comment:
Do you think there is a way for us to systematically catch all the corner cases without having implementation specific cases? Given most of these rely on values of characters consumed and bytes written, I would think, no.